src/plugins/intel_gpu/src/graph/layout_optimizer.cpp

```diff
@@ -479,6 +480,10 @@ bool layout_optimizer::can_fuse_reorder_to_prev(program_node& prev, reorder_node
         if (fmt_prev.dimension() > 6 || fmt_next.dimension() > 6)
             return false;
 
+        // Skip reorder fusing to permute when original permute output has bfxy and fusing output has bfzyx
+        if (allow_new_shape_infer && (fmt_prev.dimension() == 4) && (fmt_next.dimension() == 5) && (!node.get_users().front()->is_output()))
+            return false;
```

Comment:
I meant, we don't want the condition with hard coded conditions and output check, by using the more broader condition of allow_new_shape_infer limitation. (i.e., just allow_new_shape_infer should be enogh..)

Reply:
Got it! Updated. Thanks!